rmw_fastrtps_shared_cpp/src/types/event_types.hpp

```diff
@@ -0,0 +1,22 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TYPES__EVENT_TYPES_HPP_
+#define TYPES__EVENT_TYPES_HPP_
+
+#include "rmw/event.h"
+
+bool is_event_supported(rmw_event_type_t event_type);
```

Comment:
Why is this not in a namespace?

Reply:
APIs in `rmw_fastrtps_shared_cpp` that aren't called from `rmw_fastrtps_cpp` don't seem to be put into the `rmw_fastrtps_shared_cpp` namespace.

Reply:
Though that might be the case, I think every function should be defined in a namespace and not be global (except functions marked as `static`, which is not the case here).
No-polluting the global namespace is also recommended by the [GSG](https://google.github.io/styleguide/cppguide.html#Nonmember,_Static_Member,_and_Global_Functions).

Reply:
I have moved this function into a namespace called `rmw_fastrtps_shared_cpp_internal`.

Reply:
I'd suggest `rmw_fastrtps_shared_cpp::internal` instead.

Reply:
But I don't think you have to change it to move forward, only if you'd like to.

Reply:
I have changed to the namespace to `rmw_fastrtps_shared_cpp::internal`. Thanks.